### PR TITLE
Freestanding execution engine README

### PR DESCRIPTION
**Pull request by SabreenKaur**
This is a README.md for first time users of the freestanding-execution-engine. It takes users through an exercise of running a program in rust-exmaples inside the execution engine

### CODE REVIEW

**Comment by dominic-mulligan-arm** [comment]
Thanks @SabreenKaur for your first contribution!  Though I think there was a pull request merged into `main` concurrently with you opening this PR.  Can you rebase your changes on top of `main` on the upstream repository and force-push to get this into a state so that it's mergeable?

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 835a29acdacb456ab99fbf1c1c28370b12812c3e
* Result: FAILED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=a070a3a3-1a19-47d7-a1b0-41e88d327d70%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:1]: I'd rephrase this as "Getting started with the Veracruz freestanding execution engine", or something like that, rather than "Goal of the exercise".

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:3]: I'd rephrase this as something more direct, like "This document provides a walkthrough of how to load files...".

Also, the freestanding execution engine isn't inside a Veracruz instance: it's freestanding, to be used for debugging and development outside of a real Veracruz instance.

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:8]: Why is installing `wasmtime` necessary?  Building the freestanding execution engine should pull in the `wasmtime` dependency from `crates.io`.

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:11]: Put backticks around `read-file` too?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:17]: sp: Web assembly --> WebAssembly

There's also a tab character between "is" and "inside" (and between `read-` and `file`).

Put backticks around `main.rs`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:19]: Put backticks around `read-file.wasm`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:26]: Backticks around `info`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:29]: Backticks around `input.txt`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:34]: Backticks around `read-file`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:35]: Backticks around `stdout`?

File: sdk/freestanding-execution-engine/README.md
Comment (dominic-mulligan-arm) [inline sdk/freestanding-execution-engine/README.md:36]: sp: Read-file --> read-file.  Also put backticks around it and `rust-example`?

**Review by dominic-mulligan-arm — COMMENTED** [review]
Thanks @SabreenKaur.  Looks great, just a few suggestions for improvement.

File: sdk/freestanding-execution-engine/README.md
Comment (SabreenKaur) [reply sdk/freestanding-execution-engine/README.md:8]: I think I wrote this because I was experimenting with `wasmtime` outside of the Veracruz project as well. I will remove this part of the step.

File: sdk/freestanding-execution-engine/README.md
Comment (SabreenKaur) [reply sdk/freestanding-execution-engine/README.md]: Rephrased introduction.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: f98da8d0ea2d275be57c83337c1c43d03532f494
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=591a1fd4-45b0-4757-9290-04e282791daa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c598083f01c402e81ddc7c7bc179852d56291178
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=756033f4-048f-42f4-a3de-3130f66dd905%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c05eca9a89e4fe846ba352ad4b95755f117891f9
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=4dced239-dab7-4978-8f76-38ecc0063564%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: b7c8f85f1241c1744b2012326518a537cfdd34bc
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=e58d96c7-2807-4a50-a5f6-1e2571bfea4b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 602c2f96624cd80d16f60612c34a045f618ccede
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=671083f2-1419-4bf2-84d7-f31702545d24%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: sdk/freestanding-execution-engine/README.md
Comment (ShaleXIONG) [inline sdk/freestanding-execution-engine/README.md:10]: Probably do not need to say this line, as people use different IDE.

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
LGTM, though @SabreenKaur can you rebase onto `main` again due to @dreemkiller's changes being merged yesterday in preparation for merging?  Once you do that, I'll approve again and merge this.

**Comment by SabreenKaur** [comment]
I just rebased, though it looks like merging is still blocked, do i need to do it once again?

**Comment by dominic-mulligan-arm** [comment]
@SabreenKaur merging is blocked by the branch protection rules on `main` until your changes are even with `main`, CI successfully builds, and you have at least one vote of approval from reviewers (note my previous vote was rescinded as you pushed further changes with your rebase).  I'll approve now, and once CI builds we can merge this.

**Review by dominic-mulligan-arm — DISMISSED** [review]

**Comment by dominic-mulligan-arm** [comment]
@SabreenKaur can you rebase again?  Sorry, we have a big backlog of merge requests that are ready/almost ready (including this one).  Let's get this one in next, so it's out of the way, as it's just documentation.

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: af8311a3b0818112603515f5537aa5e282fe157e
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=967949b6-c22f-4481-9f18-7b358c9463d9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by dominic-mulligan-arm** [comment]
Oh wow, CI is very fast now.

+1 reached, merging.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: af8311a3b0818112603515f5537aa5e282fe157e
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=185ada0c-9a7d-47a1-9ad9-ff48e4643b14%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 23f91feaae82ee54dd6b92543e45058ef58108ac
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=b587a84e-8419-4d2d-a8fe-2a4668d8d95d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 40a9c3bf1b60821200bf9310a3ae8f0d51c40d07
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=fd95c436-b8bd-49c4-9f5e-1eb1b0f19061%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**
